flink-table-store-core/src/main/java/org/apache/flink/table/store/table/ReadonlyTable.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.store.table;
+
+import org.apache.flink.table.store.table.sink.TableCommit;
+import org.apache.flink.table.store.table.sink.TableWrite;
+
+/** Readonly table which only provide implementation for scan and read. */
+public interface ReadonlyTable extends Table {
+
+    @Override
+    default TableWrite newWrite(String commitUser) {
+        throw new UnsupportedOperationException(
+                String.format(
+                        "Readonly Table %s does not support newWrite.",
```

Comment:
No need to tell users the class name of `ReadonlyTable`? Or provide user with the type of table?

Reply:
I think it is reasonable to provide class name.